src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -43,24 +43,66 @@ ov::Tensor Bank::get(const ov::Tensor& tensor) {
     return tensor;
 }
 
-std::shared_ptr<Bank> BankManager::getBank(const std::string& bank_name) {
+ov::Tensor Bank::get(const ov::Tensor& tensor, const std::string& device) {
+    if (!tensor) {
+        OPENVINO_THROW("Uninitialized tensor in weights bank allocation!");
+    }
+
+    if (device != "CPU" && device != "NPU") {
+        OPENVINO_THROW("Unsupported device in weights bank allocation: ", device);
+    }
+
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    auto iter_cpu = m_bank.find(tensor.data());
+    if (iter_cpu == m_bank.end()) {
+        OPENVINO_THROW("Unknown tensor in weights bank allocation!");
+    }
+
+    // If target device is CPU - just reuse the default bank
+    if (device == "CPU") {
+        return iter_cpu->second;
+    }
+
+    // Non-CPU - check if the tensor is already there
+    auto& device_bank = m_device_bank[device];
+    auto iter_device = device_bank.find(tensor.data());
+    if (iter_device != device_bank.end()) {
+        // Already allocated on the device - reuse
+        return iter_device->second;
+    }
+
+    // Allocation needed
+    // FIXME: set the context in the constructor
+    if (device == "NPU" && !m_remote_ctx) {
+        m_remote_ctx = m_plugin->get_core()->get_default_context(device)._ptr;
+    }
```

Comment:
it seems you need `m_plugin` only to access `ov::Core`? If you do this to avoid creating multiple `ov::Core`'s maybe that's `ov::Core` is what you need to pass when you initialize the bank?

I believe we can multiple ov::Core objects in the app. Would it work then? What if app written in such way that there's two individual `ov::Core` objects created for two models which share weights?

Probably you need to clarify it with @ilya-lavrenov

Reply:
```
 * User applications can create several Core class instances, but in this case the underlying plugins
 * are created multiple times and not shared between several Core instances. The recommended way is to have
 * a single Core instance per application.
```

Reply:
Done